reactioncommerce/reaction-next-starterkit
# refactor: 248 nnnnat checkout login

Resolves #248 
Impact: **major**
Type: **refactor**

## Issue
When a customer enters checkout without an email or account they'll be redirected to a login page. This happens on the client and can take a few seconds causing a flash of the Checkout before the redirect.

## Solution
 * Updated checkout page component to check if a customer has an account/email in a `componentDidMount` lifecycle method and replace `Router.pushRoute` with `Router.replaceRoute`. Using `.replaceRoute` should preserve the intended back button behavior.
 * Updated login page component to also check if a customer has an account/email in `componentDidMount`.
 * Fixed import paths in a few component tests.
 * Fixed a "missing key" warning in _document `<Head />`

## Breaking changes
N/A

## Testing
1. As an anonymous customer select a product and add it to the cart.
2. Navigate to the `/checkout` page.
    * The page should redirect to the `/login` page.
    * There should be no flash of the "Checkout" view before the redirect happens.
3. Click the browsers back button.
    * The storefront should return to the last page before navigating to the `/checkout` page.
